xtable-hive-metastore/src/test/java/org/apache/xtable/hms/TestHMSCatalogSyncClient.java

```diff
@@ -72,7 +76,8 @@ private HMSCatalogSyncClient createHMSCatalogSyncClient() {
         mockHMSCatalogConfig,
         testConfiguration,
         mockMetaStoreClient,
-        mockTableBuilder);
+        mockTableBuilder,
+        Optional.of(partitionSyncTool));
```

Comment:
Can you have a case where this is empty as well to show that it does not cause issues?